### DISCUSSION TITLE
How do we add missing elements?

### DISCUSSION THREAD

**Discussion by mapembert** [IBM/carbon-ui-builder#186] [General]
This is a cool utility and we may have our designers use it to build out requirements for new applications.  It is missing some elements/components like the data grid. Can someone show us how to add this 'element' to the utility with a **step by step** please?

Referencing the Carbon Design System:
https://carbondesignsystem.com/components/overview/

**Comment by zvonimirfras** [comment]
Hi Mike! (@mapembert)

🔍  General answer to this is [open a feature request](https://github.com/IBM/carbon-ui-builder/issues/new/choose).

🚀  If you're willing to contribute it, comment on the issue to say so and then we'll talk on slack, discord or wherever to get you up to speed and contributing.

---

For [table](https://carbondesignsystem.com/components/data-table/usage/) in particular - this is a _very_ complex component and we'll probably have to do it in stages, or it will take a long while.

Which features from the table do you think you need to be useful?

_You can find me on slack, discord (Zvonimir#9989) or anywhere else, I'm easy to find._

**Reply by mapembert** [reply]
Thank you for the reply @zvonimirfras! 

IBM owns Maximo and TRIRIGA and I specialize in TRIRIGA. They released support for Polymer 1, 3 and now React.  In the shipped applications IBM creates they use the carbon design so this tool is really applicable to us.  We are learning react currently on our team and so I wouldn't be able to contribute yet as I don't know what I'm doing.

Regarding your question, for a table we would need to be able to use it just like methods we use in web components. Here is an example:
https://github.com/vaadin/vaadin-grid/tree/v2.0.0-alpha4

Basically, bind an array of objects, specify the properties you want to make into columns and it renders all the rows.

We can use 3rd party software also but the carbon design is not applied so I thought I would ask for it. I understand it is a lot of work and I don't want to burden you.  We can just insert a place holder where we want to show a list or table for now.

Thank you!
